how-to-release.md

```diff
@@ -0,0 +1,274 @@
+
+## Step 1: Tagging
+```
+# Checkout to the latest successful-voted tag
+git checkout v0.3.0-rc2
+# Tag this existing tag again as a formal release tag `v0.3.0` and push
+git tag v0.3.0
+git push origin v0.3.0
+```
+
+## Step 2: Create the release page on GitHub
+
+On the GitHub repository page, go to "Releases" and create a new draft. On the draft, fill out the tag field with the correct tag and do NOT check the pre-release box. There are two sections needs to be filled in this github page.
+- Release notes
+- Release assets
+
+### Release notes
+Some introductory text, highlighting the important changes going in the release. This should be reviewed earlier by the community.
+
+### Release assets
+
+Run the following commands to build the distribution artifacts
+```
+./gradlew clean distribution
+```
+
+The files resulting from the build will be under `build/distributions`. For each one of the `.zip`, `.tgz` or `.jar` files in that directory, generate checksums (currently `md5`, `sha1`, and `sha256`). It is easy to do with this bash script.
+```
+#!/bin/bash
+
+for file in ./*.tgz ; do md5 $file > $file.md5 ; done
+for file in ./*.tgz ; do shasum -a 1 $file > $file.sha1 ; done
+for file in ./*.tgz ; do shasum -a 256 $file > $file.sha256 ; done
+for file in ./*.tgz ; do gpg --armor --output $file.asc --detach-sig $file ; done
+
+for file in ./*.zip ; do md5 $file > $file.md5 ; done
+for file in ./*.zip ; do shasum -a 1 $file > $file.sha1 ; done
+for file in ./*.zip ; do shasum -a 256 $file > $file.sha256 ; done
+for file in ./*.zip ; do gpg --armor --output $file.asc --detach-sig $file ; done
+
+for file in ./*.jar ; do md5 $file > $file.md5 ; done
+for file in ./*.jar ; do shasum -a 1 $file > $file.sha1 ; done
+for file in ./*.jar ; do shasum -a 256 $file > $file.sha256 ; done
+for file in ./*.jar ; do gpg --armor --output $file.asc --detach-sig $file ; done
+```
+Upload all these files and its checksums onto the "Assets" section.
+
+Finally, double check all the content is correct and then you can click the "Publish" button to publish the formal github release.
+
+## Step 3: Release the artifact to maven central
+
+Choose the latest release candidate in sonatype and click on the "Release" button in the top menu bar to release the artifact. Wait until it shows up in Maven Central, it takes some [time](http://central.sonatype.org/pages/ossrh-guide.html#SonatypeOSSMavenRepositoryUsageGuide-9.ActivateCentralSync).
+
+## Step 4: Documentation
+
+Generate the documentation and update onto the website.
+
+## Step 5: Push the images to Docker Hub
+For this step, you need a Docker Hub account associated to Pravega.
+
+Once you are ready, run the following steps:
+
+- `docker login` (using a dockerhub account with write access)
+- `./gradlew clean docker dockerPush` (this pushes the specific version tags)
```

Comment:
ideally it should be better to not "rebuild" the artifacts, but we should push the artifacts we built during the VOTE.

it would be better to include the docker images in the VOTE, as many users (think about k8s) use the docker images

btw this is not the current practice in Pravega, and also in all the projects I am involved in the ASF we never vote on the docker images as part of the process, but they are always generated after the VOTE has been closed

Reply:
I also agree with that because for my validation in the past, I sometimes also need to build docker images from code to test some samples work. Besides, we also had some regressions in the past in dockerized standalone, which can be partly blamed by not including the docker images in VOTE phase.

Reply:
I would prefer to make the RC docker images in the release manager's personal repositories. In this way, we can still verify it as a public image, and we can get rid of the long list of redundant images and perhaps the tag order conflict in the official `pravega` dockerhub account.